freyja/read_analysis_tools.py

```diff
@@ -0,0 +1,268 @@
+import os
+import re
+import pysam
+
+
+def extract(query_mutations, input_bam, output):
+    # Load data
+    with open(query_mutations) as infile:
+        mutations = infile.read().splitlines()
+
+        if len(mutations) < 3:
+            for i in range(3 - len(mutations)):
+                mutations.append('')
+        snps = [s.strip() for s in mutations[0] .split(',')]
+        insertions = [s.strip() for s in mutations[1].split(',')]
+        deletions = [s.strip() for s in mutations[2].split(',')]
+    try:
+        # parse tuples from indels
+        if len(insertions[0]) > 0:
+            insertions = [
+                (int(s.split(':')[0][1:]), s.split(':')[1][1:-2].strip('\''))
+                for s in insertions
+            ]
+
+        if len(deletions[0]) > 0:
+            deletions = [
+                (int(s.split(':')[0][1:]), int(s.split(':')[1][:-1]))
+                for s in deletions
+            ]
+
+        # get loci for all mutations
+        snp_sites = [int(m[1:len(m)-1])-1 for m in snps if m]
+        indel_sites = [s[0] for s in insertions if s] +\
+            [s[0] for s in deletions if s]
+    except ValueError:
+        print('extract: Error parsing', query_mutations)
+        print('extract: See README for formatting requirements.')
+        return -1
+
+    snp_dict = {int(mut[1:len(mut)-1])-1: mut[-1] for mut in snps if mut}
+
+    samfile = pysam.AlignmentFile(input_bam, 'rb')
+
+    reads_considered = []
+
+    all_sites = snp_sites + indel_sites
+    all_sites.sort()
+
+    for site in all_sites:
```

Comment:
worth noting that this is can be somewhat redundant, since neighboring sites will almost always be on the same reads

Reply:
The lazy way I'd had it before, where you just grab everything sort of solves this... but then you get stuff in between that you don't want of course. The ideal way to do this would be to use the primer bed file, but that's more than we need at this stage.